src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -190,43 +190,94 @@ namespace
           : m_renderer(renderer)
           , m_active_camera(camera)
         {
+            m_callback_key = GetISceneEventManager()->RegisterCallback(this, false, 100, true);
         }
 
-        virtual void ControllerOtherEvent(NodeKeyTab& nodes) override
+        virtual ~SceneChangeCallback()
+        {
+            GetISceneEventManager()->UnRegisterCallback(m_callback_key);
+        }
+
+        virtual void ModelOtherEvent(NodeKeyTab& nodes) override
         {
             if (m_active_camera == nullptr || m_renderer == nullptr)
                 return;
-            
+
             for (int i = 0, e = nodes.Count(); i < e; ++i)
             {
                 if (NodeEventNamespace::GetNodeByKey(nodes[i]) == m_active_camera)
                 {
-                    m_renderer->update_camera_transform(m_active_camera);
+                    m_renderer->update_camera_object(m_active_camera);
                     m_renderer->get_render_session()->reininitialize_render();
                     break;
                 }
             }
         }
 
-        virtual void ModelOtherEvent(NodeKeyTab& nodes) override
+      private:
+        SceneEventNamespace::CallbackKey    m_callback_key;
+        AppleseedInteractiveRender*         m_renderer;
+        INode*                              m_active_camera;
+    };
+
+    class ViewportCallback 
+      : public RedrawViewsCallback
+    {
+      public:
+        ViewportCallback()
+          : m_current_view(nullptr)
+          , m_last_fov(0)
+          , m_last_timer(0)
```

Comment:
`nullptr`

Reply:
It doesn't accept nullptr - says nullptr cannot be used to initialize UINT_PTR